library/proc_macro/src/bridge/buffer.rs

```diff
@@ -78,18 +78,40 @@ impl<T: Copy> Buffer<T> {
         mem::take(self)
     }
 
+    pub(super) fn extend_from_array<const N: usize>(&mut self, xs: &[T; N]) {
+        if xs.len() > self.capacity.wrapping_sub(self.len) {
```

Comment:
```suggestion
        if xs.len() > self.capacity - self.len {
```
Isn't capacity guaranteed to always be larger or equal than length?
(Here and one more case below.)

Reply:
Ah. Yeah, seems correct -- I had taken this code from raw_vec's [needs_to_grow](https://github.com/rust-lang/rust/blob/9f75dbfa69ba8508fa7765305f75cd96dff06078/library/alloc/src/raw_vec.rs#L400) and I'm not actually sure it's necessary there either, but it shouldn't be necessary here.